pep-9999.rst

```diff
@@ -0,0 +1,1037 @@
+     prevent this`__. So, for those distributions, this PEP
+     is not a behavior change; it simply standardizes that
+     behavior in a way that is no longer Debian-specific and
+     can be included into upstream pip.
+
+     .. __: https://sources.debian.org/src/python-pip/20.3.4-2/debian/patches/hands-off-system-packages.patch/
+
+     (We have seen user reports of externally-installed
+     packages being deleted on Debian or a derivative. We
+     suspect this is because the user has previously run
+     ``sudo pip install --upgrade pip`` and therefore now
+     has a version of ``/usr/bin/pip`` without the Debian
+     patch; standardizing this behavior in upstream package
+     installers would address this problem.)
+
+3. A distribution Python when used inside a virtual environment.
+
+   * Can you ``pip install`` without an override? Currently
+     yes; after this PEP yes, because it specifically
+     exempts virtual environments.
+
+   * Can it delete externally-installed packages? There are
+     no externally-installed packages in a virtual
+     environment, so this question is irrelevant - all
+     packages in the environment can be deleted. Even when
+     ``pip``, ``setuptools``, etc. are installed into the
+     environment, they are and should be locally-managed;
+     they are not system-managed.
+
+4. A distribution Python when used inside a virtual
+   environment with ``--system-site-packages``. This is like
+   the previous case, but worth calling out explicitly,
+   because anything on the global ``sys.path`` is visible.
+
+   * Can you ``pip install`` without an override? Currently
+     yes; after this PEP yes.
+
+   * Can it delete externally-installed packages? Currently
+     no, because pip has a special case for running in a
+     virtual environment and deleting packages outside it.
+     After this PEP, still no, in a more general way,
+     because system site packages will be outside any of the
+     ``sysconfig`` schemes for the environment.
+
+5. A distribution Python when used in a single-application
+   container image (e.g., a Docker container). In this use
+   case, the risk of breaking system software is lower,
+   since generally only a single application runs in the
+   container, and the impact is lower, since you can rebuild
+   the container and you don't have to struggle to recover a
+   running machine. There are also a large number of
+   existing ``Dockerfile``s with an unqualified ``RUN pip
+   install ...`` statement, etc., and it would be good not
+   to break those.  So, builders of base container images
+   may want to ensure that the marker file is not present,
+   even if the underlying OS ships one by default. In such a
+   case:
+
+   * Can you ``pip install`` without an override? Currently
+     yes; after this PEP yes.
+
+   * Can it delete externally-installed packages? Currently
+     yes; after this PEP no. This is a behavior change which
```

Comment:
Why is this particular case yes/no? I don't understand how this is different than the regular distro case.

Reply:
Not sure I follow - it is not different from the regular distro case (case 2), which is also yes->no.

The behavior change between case 2 and case 4 is in the "Can you pip install" question, which is yes->no for case 2 and yes->yes for case 4.

Reply:
Can we just turn this section into like... a giant table?

Reply:
That's what I was originally going to do but there's a lot of explanation on each one. Maybe table up front, rationale below?

Reply:
I added a table, but I think I managed not to actually reduce the explanation below by very much. :) Still, it should be more readable, take a look?